### PR TITLE
[Resources.OperatingSystem] initial implementation

### LINKED ISSUE #1844: Provide an operating system resource package
### Component

None

### Is your feature request related to a problem?

I want information about my os to be visible in my open Telemetry data

### What is the expected behavior?

The following properties are set unless a user opts out

- os.build_id
- os.description
- os.name
- os.type

### Which alternative solutions or features have you considered?

Manually adding them

### Additional context

https://opentelemetry.io/docs/specs/semconv/resource/os/

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes #1844

## Changes

[Resources.OperatingSystem] initial implementation

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Resources.OperatingSystem/OperatingSystemSemanticConventions.cs
Comment (cijothomas) [inline src/OpenTelemetry.Resources.OperatingSystem/OperatingSystemSemanticConventions.cs:10]: this does not look like is needed. If needed for tests, its best to declare it in tests project.

Comment (Kielek) [reply src/OpenTelemetry.Resources.OperatingSystem/OperatingSystemSemanticConventions.cs:10]: I would say that you should define here separate values of the resource values.

You can see example here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/49a01d3a8b72cbdba499d3bac302e055209eb89a/src/OpenTelemetry.SemanticConventions/Attributes/CloudAttributes.cs#L73-L81

File: test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs
Comment (cijothomas) [inline test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs:13]: I'd suggest to use the extension method instead of using the ctor, so as to mimic the user code more closely, and also this can help remove the InternalsVisible trick as well. There is no need for Tests to know any internals.

Comment (cijothomas) [reply test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs]: Won't be a bad idea at all to copy the semantic convention OR add ref to it from test project.

Comment (Kielek) [reply test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs]: @cijothomas, in general I agree, but current approach is in sync with all other projects.
If we want to change lets do it in separate PR/issue.

Comment (cijothomas) [reply test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs]: yes agree, no need to block this PR at all.

**Review by Kielek — APPROVED** [review]
@open-telemetry/dotnet-contrib-maintainers, if you are fine, I wuld like to merge this and make `v0.1.0-alpha.1` release tomorrow.

File: src/OpenTelemetry.Resources.OperatingSystem/README.md
Comment (cijothomas) [inline src/OpenTelemetry.Resources.OperatingSystem/README.md:24]: nit: Resource can be used with all signals, not just TracerProvider.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**
